java/src/jmri/ThrottleManager.java

```diff
@@ -35,154 +35,61 @@
-     * @return true if the request will continue, false if the request will not
-     *         be made; false may be returned if a the throttle is already in
-     *         use
-     */
-    public boolean requestThrottle(LocoAddress address, BasicRosterEntry re, ThrottleListener l);
-
-    /**
-     * Cancel a request for a throttle.
-     * <P>
-     * This is a convenience version of the call, which uses system-specific
-     * logic to tell whether the address is a short or long form.
-     *
-     * @param re desired Roster Entry
-     * @param l  ThrottleListener canceling the request for a throttle
-     */
-    public void cancelThrottleRequest(BasicRosterEntry re, ThrottleListener l);
-
-    /**
-     * Cancel a request for a throttle.
-     * <P>
-     * This is a convenience version of the call, which uses system-specific
-     * logic to tell whether the address is a short or long form.
-     *
-     * @param address desired decoder address
-     * @param l       ThrottleListener canceling request for a throttle
-     */
-    public void cancelThrottleRequest(int address, ThrottleListener l);
-
+    public boolean requestThrottle(LocoAddress address, ThrottleListener l , boolean canHandleDecisions);
+    
     /**
      * Cancel a request for a throttle.
      *
-     * @param address desired decoder address
-     * @param isLong  true if requesting a DCC long (extended) address
+     * @param address unwanted Loco address
      * @param l       ThrottleListener canceling request for a throttle
      */
-    public void cancelThrottleRequest(int address, boolean isLong, ThrottleListener l);
+    public void cancelThrottleRequest(LocoAddress address, ThrottleListener l);
 
     /**
-     * Steal a requested throttle.
-     * <P>
-     * This is a convenience version of the call, which uses system-specific
-     * logic to tell whether the address is a short or long form.
+     * Steal or Share a requested throttle.
      *
-     * @param re desired Roster Entry
-     * @param l  ThrottleListener requesting the throttle steal occur.
-     * @param steal true if the request should continue, false otherwise.
+     * @param address desired LocoAddress
+     * @param l The ThrottleListener which has made the decision
+     * @param decision from the ThrottleListener, STEAL or SHARE.
      * @since 4.9.2
      */
-    public void stealThrottleRequest(BasicRosterEntry re, ThrottleListener l,boolean steal);
-
-    /**
-     * Steal a requested throttle.
-     * <P>
-     * This is a convenience version of the call, which uses system-specific
-     * logic to tell whether the address is a short or long form.
-     *
-     * @param address desired decoder address
-     * @param l  ThrottleListener requesting the throttle steal occur.
-     * @param steal true if the request should continue, false otherwise.
-     * @since 4.9.2
-     */
-    public void stealThrottleRequest(int address, ThrottleListener l,boolean steal);
```

Comment:
Please do not remove the methods that use an integer to interact with a throttle.  Doing this makes the API less flexible.

Reply:
Thanks for the guidance, have retained